lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -258,14 +248,150 @@ def _prepare_to_run(self, params, concierge_params=None) -> JobSubmissionParamet
             source_ws_objects=params.get(_SOURCE_WS_OBJECTS),
         )
 
+    def _prepare_to_run(self, params, concierge_params=None) -> JobSubmissionParameters:
+        """
+        Creates a job record and creates the job submission params
+        """
+
+        job_id = self._init_job_rec(self.sdkmr.get_user_id(), params)
+        self.logger.debug(
+            f"User {self.sdkmr.get_user_id()} attempting to run job {params[_METHOD]} {params}"
+        )
+        return self._generate_job_submission_params(job_id, params)
+
+    def _run_multiple(self, runjob_params):
+        """
+        Get the job records, bulk save them, then submit to condor.
+        If any condor submission fails, abort all of the jobs
+        :return:
+        """
+        # Save records to db
+        init_job_rec = time.time()
+        job_records = []
+        for runjob_param in runjob_params:
+            job_records.append(
+                self._init_job_rec(self.sdkmr.get_user_id(), runjob_param, save=False)
+            )
+        self.logger.debug(f"init_job_rec = {time.time() - init_job_rec}")
+
+        save_jobs = time.time()
+        job_ids = self.sdkmr.save_jobs(job_records)
+
+        self.logger.debug(f"save_jobs = {time.time() - save_jobs}")
+        self.logger.debug(f"init and save save_jobs = {time.time() - init_job_rec}")
+
+        # Generate job submission params
+        gen_sub_time = time.time()
+        job_submission_params = []
+        for i, job_id in enumerate(job_ids):
+            job_submission_params.append(
+                self._generate_job_submission_params(job_id, runjob_params[i])
+            )
+            assert job_id == job_submission_params[i].job_id
+
+        self.logger.debug(
+            f"init and gen_sub_time save_jobs = {time.time() - gen_sub_time}"
+        )
+
+        # Takes 2.5200018882751465 for 100, can shave off 2.5 secs by making this async
+        kafku = time.time()
+        for job_id in job_ids:
+            self.sdkmr.get_kafka_client().send_kafka_message(
+                message=KafkaCreateJob(
+                    job_id=str(job_id), user=self.sdkmr.get_user_id()
+                )
+            )
+
+        self.logger.debug(f"kafka submit = {time.time() - kafku}")
+
+        # Submit to Condor
+        condor_time = time.time()
+        try:
+            submission_ids = self._submit_multiple(job_submission_params)
+            self.logger.debug(f"condor_time submit = {time.time() - condor_time}")
+            return submission_ids
+        except Exception as e:
+            self._abort_multiple_jobs(job_ids)
+            raise e
+
+    def _update_to_queued_multiple(self, job_ids, scheduler_ids):
+        # TODO Unused
```

Comment:
I'm not sure what this comment is referring to?

Reply:
removed

Reply:
:+1: